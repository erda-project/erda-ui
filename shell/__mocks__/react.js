// Copyright (c) 2021 Terminus, Inc.
//
// This program is free software: you can use, redistribute, and/or modify
// it under the terms of the GNU Affero General Public License, version 3
// or later ("AGPL"), as published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

const react = require('react');
// Resolution for requestAnimationFrame not supported in jest error :
// https://github.com/facebook/react/issues/9102#issuecomment-283873039
global.window = global;
window.addEventListener = () => {};
window.requestAnimationFrame = () => {
  // throw new Error('requestAnimationFrame is not supported in Node');
};

module.exports = react;
