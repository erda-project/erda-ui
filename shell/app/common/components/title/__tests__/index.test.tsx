// Copyright (c) 2021 Terminus, Inc.
//
// This program is free software: you can use, redistribute, and/or modify
// it under the terms of the GNU Affero General Public License, version 3
// or later ("AGPL"), as published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import Title from '..';

describe('Title', () => {
  const title = 'erda title';
  const tips = 'erda title';
  it('should render well', async () => {
    const result = render(<Title title={title} />);
    expect(result.container).isExist('.h-12', 1);
    expect(result.getByText(title)).toBeTruthy();
    expect(result.container).isExist('[name="help"]', 0);
    expect(result.container).isExist('.border-bottom', 1);
    result.rerender(<Title title={title} tip={tips} />);
    expect(result.container).isExist('[name="help"]', 1);
    userEvent.hover(result.container.querySelector('[name="help"]')!);
    await waitFor(() => expect(screen.queryByText(tips)).toBeInTheDocument());
    result.rerender(<Title title={title} showDivider={false} />);
    expect(result.container).isExist('.border-bottom', 0);
  });
  it('should render with operations', () => {
    const result = render(
      <Title title={title} mb={16} mt={8} operations={['title1', { title: 'title2' }, <div>title3</div>]} />,
    );
    expect(result.getByText('title2')).toBeTruthy();
    expect(result.getByText('title3')).toBeTruthy();
  });
});
