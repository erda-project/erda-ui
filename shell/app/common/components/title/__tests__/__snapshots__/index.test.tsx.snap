// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should  1`] = `
<div>
  <div
    class="ec-title w-full flex items-center justify-between h-12 mb-4  border-bottom"
  >
    <div
      class="inline-flex items-center font-medium mr-2"
    >
      <div
        class="text-lg"
      >
        erda title
      </div>
    </div>
    <div
      class="flex-1 inline-flex items-center justify-end"
    />
  </div>
</div>
`;
